### DISCUSSION TITLE
X2RAY TLS

### DISCUSSION THREAD

**Discussion by aydinj** [Ysurac/openmptcprouter#3256] [Q&A]
Can OpenMPTCPRouter change or set TLS with a certificate, SNI, and other Obfuscation tools for v2ray or x2ray? For example, using GUI or config files. I couldn't find any guide about it in wiki.

**Comment by Ysurac** [comment]
OpenMPTCProuter can use Xray with VLESS Reality on port 443 for example. Else Trojan, VLESS and VMESS are also available.

**Reply by aydinj** [reply]
Thank you very much for the response.
But I was asking about customizing VMESS or VLESS using your own SSL Certificate and custom SNI and port. I know it is unavailable in GUI, But I think changing the x2ray core config file could be possible. I don't know where it is located or if it will cause conflict with other services on the server.

https://xtls.github.io/en/config/transport.html

**Reply by Ysurac** [reply]
It's in ```/etc/xray``` on the VPS and via "uci" command for router part.
Some changes are possible only user and encryption algo are managed via API.

**Reply by aydinj** [reply]
Thank you. In the server, I could modify "xra-vless-reality.json". It has two parameters that I need to change:
_"realitySettings":{"dest"}_ and another  was _{"serverNames"}_
The first is the VLESS address, and the second is for the obfuscation of the TLS SNI.

In the router, I also need to modify them. However, usnig UCI I could only modify _{xray.omrout.s_vless_reality_address='1.1.1.1:443'}_ but I couldn't find "severNames" 
Is it possible to modify "serverNames" in the router by the user? For example, an option should be named: _"xray.omrout.s_vless_reality_serverNames"_
This option is essential for networks that use DPI firewalls to filter VPNs.

**Reply by Ysurac** [reply]
I just added in init script support for s_vless_reality_fingerprint, s_vless_reality_servername and s_vless_reality_spiderx settings on router side.
New snapshot is compiling, this will be available in next release.
